docs/modules/reference/pages/configuration/filters/filters.adoc

```diff
@@ -0,0 +1,70 @@
+
+[[filters]]
+= Filters
+
+A filter (also known as a rule) is a shorthand SQL statement that automatically selects ipaddr from the ipinterface table. 
+The rule you write builds the WHERE clause. 
+The https://github.com/OpenNMS/opennms/blob/develop/opennms-config/src/main/resources/database-schema.xml[database-schema.xml] file in the /opt/opennms/etc directory informs the filter code the tables it can use to build the WHERE clause.
+
+== Explanation of database-schema.xml file
+
+Each table the filter code uses appears in a <table> tag. 
+If a table has an attribute of "visible=false", then none of the columns in that table can be used in the WHERE clause and thus cannot appear in the rule. 
+You will get a syntax exception if it sees any non-visible columns in the rule. 
+The same applies to a non-visible column in a table.
+
+A <join> tag tells the filter module how to relate this table to the ipinterface table. 
+The ipinterface table is marked with the attribute "key=primary", meaning that it is going to be the table we select ipaddr from and join all other tables to.
+
+=== Operators
+
+You can use C/Java-style comparison operators with data types they apply to.
+(You can use == and != on strings, as well as the SQL "LIKE" keyword.)
+
+For LIKE comparisons, the character "_" matches any single character and "%" matches any series of characters (or none at all). 
+For example, "F_o%"" matches "Foo", "Foom", and "Flowers" but not "Foip".
+
+To handle NULL values (which include cases where you've joined across to a table where there is no matching row), use the "IS NULL" and "IS NOT NULL" operators. 
+Note that comparing a null value to anything with any other operator always returns false, so 
+`categoryName != 'SomeCategory'` will not return anything with a null categoryName. 
+Instead, you would need to use `categoryName != 'SomeCategory' | categoryName IS NULL` to exclude all nodes that are tagged with `SomeCategory`.
+
+You can use parentheses to group expressions and can apply boolean operators anywhere in an expression. 
+Note that, in a departure from C/Java convention, boolean operators are single characters rather than double, so they look more like the bitwise arithmetic operators in C:
```

Comment:
```suggestion
NOTE: In a departure from C/Java convention, boolean operators are single characters rather than double, so they look more like the bitwise arithmetic operators in C:
```